packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -219,7 +219,9 @@ const Combobox = forwardRef<HTMLInputElement, ComboboxProps>(
 
     const handleComboboxOptionClick = useCallback(() => {
       // maintain focus on the input
-      inputRef.current?.focus();
+      if (inputRef.current !== document.activeElement) {
+        inputRef.current?.focus();
+      }
```

Comment:
This was previously causing a test to fail for in rtl, but we should only try to focus the element again when the element doesn't already have focus.